src/mlx/warnings.py

```diff
@@ -165,15 +188,19 @@ def warnings_wrapper(args):
     parser.add_argument('--minwarnings', type=int, required=False, default=0,
                         help='Minimum amount of warnings accepted')
     parser.add_argument('--version', action='version', version='%(prog)s {version}'.format(version=pkg_resources.require('mlx.warnings')[0].version))
-
+    parser.add_argument('--config', dest='configfile', action='store', required=False, help='Config file in JSON format provides toggle of checkers and their limits')
     parser.add_argument('logfile', nargs='+', help='Logfile (or command) that might contain warnings')
     parser.add_argument('flags', nargs=argparse.REMAINDER, help='Possible not-used flags from above are considered as command flags')
 
     args = parser.parse_args(args)
 
-    warnings = WarningsPlugin(sphinx=args.sphinx, doxygen=args.doxygen, junit=args.junit, verbose=args.verbose)
-    warnings.set_maximum(args.maxwarnings)
-    warnings.set_minimum(args.minwarnings)
+    # Read config file
+    if args.configfile is not None:
+        warnings = WarningsPlugin(configfile=args.configfile)
+    else:
+        warnings = WarningsPlugin(sphinx=args.sphinx, doxygen=args.doxygen, junit=args.junit, verbose=args.verbose)
+        warnings.set_maximum(args.maxwarnings)
+        warnings.set_minimum(args.minwarnings)
```

Comment:
this means those levels only count when you don't pass a config file. If we indent one level less, it can serve as global min/max levels. While config file will contain min/max per checker.

Reply:
We dont have global min/max levels as we have per checker checks. I would leave it this way

Reply:
then maybe include this - together with d/s/j argument -  in the mutual exclusion to config argument?

Reply:
Did that indeed. See 6 lines above :)